### PR TITLE
Fixes 'Sliders dont appear in dark mode in the Gallery' issue

### PR DESCRIPTION
**Pull request by msal4**
A fix for issue #27266

### CODE REVIEW

File: examples/flutter_gallery/lib/gallery/themes.dart [hunk 1]
```diff
@@ -43,6 +43,12 @@ ThemeData _buildDarkTheme() {
       colorScheme: colorScheme,
       textTheme: ButtonTextTheme.primary,
     ),
+    sliderTheme: SliderThemeData.fromPrimaryColors(
+      primaryColor: primaryColor,
+      primaryColorDark: primaryColor,
```
Comment (clocksmith) [inline examples/flutter_gallery/lib/gallery/themes.dart]: For the time being, let's use secondaryColor for primaryColorLight, and Color(0xFF0050a0) for primaryColorDark

**Review by clocksmith — CHANGES_REQUESTED** [review]
LGTM, but we can use different colors instead of just primaryColor

**Review by clocksmith — APPROVED** [review]
LGTM

**Comment by clocksmith** [comment]
Thanks for fixing this! After looking further into this, it might make more sense to actually override these colors in the colorScheme (or theme) for the dark theme builder. Does that work?

**Comment by msal4** [comment]
@clocksmith Thanks for reviewing the code, I tried changing it in the color scheme and theme data but there is no option for overriding the primary color for the sliders.
If there is a better way of doing it, please do let me know.

**Comment by HansMuller** [comment]
What I think @clocksmith is suggesting is replacing this line from _buildDarkTheme() in examples/flutter_gallery/lib/gallery/themes.dart

```
  final ThemeData base = ThemeData.dark();
```

With:

```
  final ThemeData base = ThemeData(
    brightness: Brightness.dark,
    primaryColorDark: const Color(0xFF0050a0),
    primaryColorLight: secondaryColor,
  );
```

You could probably just flatten both _buildFooTheme() methods, like:

```
ThemeData _buildDarkTheme() {
  const Color primaryColor = Color(0xFF0175c2);
  const Color secondaryColor = Color(0xFF13B9FD);
  final ColorScheme colorScheme = const ColorScheme.dark().copyWith(
    primary: primaryColor,
    secondary: secondaryColor,
  );
  return ThemeData(
    brightness: Brightness.dark,
    primaryColor: primaryColor,
    primaryColorDark: const Color(0xFF0050a0),
    primaryColorLight: secondaryColor,
    buttonColor: primaryColor,
    indicatorColor: Colors.white,
    accentColor: secondaryColor,
    canvasColor: const Color(0xFF202124),
    scaffoldBackgroundColor: const Color(0xFF202124),
    backgroundColor: const Color(0xFF202124),
    errorColor: const Color(0xFFB00020),
    buttonTheme: ButtonThemeData(
      colorScheme: colorScheme,
      textTheme: ButtonTextTheme.primary,
    ),
    textTheme: _buildTextTheme(base.textTheme),
    primaryTextTheme: _buildTextTheme(base.primaryTextTheme),
    accentTextTheme: _buildTextTheme(base.accentTextTheme),
  );
}
```

**Comment by msal4** [comment]
https://github.com/flutter/flutter/blob/a6e901130e08c2b8000001af1a4c13165d32321a/packages/flutter/lib/src/material/slider.dart#L109
Overriding the `primaryColorDark` and `primaryColorLight` in `ThemeData` does not affect the sliders, if we want to override the primary colors, we either pass it to the `Slider` constructor (activeColor and inactiveColor) or we change it in the ThemeData using sliderThemeData.
We could also change this behaviour in the `Slider` itself.

**Comment by HansMuller** [comment]
I've moved this fix to #28619

**Pull request closed without merging**